src/github_sdk.py

```diff
@@ -1,101 +1,134 @@
 import gzip
 import io
 import logging
-import os
 import uuid
 from datetime import datetime
 
 import requests
 from sentry_sdk.envelope import Envelope
 from sentry_sdk.utils import format_timestamp
 
-logging.basicConfig(level=os.environ.get("LOGGING_LEVEL", "INFO"))
-
-# We need an authorized token to fetch the API. If you have SSO on your org you will need to grant permission
-# Your app and the Github webhook will share this secret
-# You can create an .env file and place the token in it
-GH_TOKEN = os.environ.get("GH_TOKEN")
-# Where to report Github actions transactions
-SENTRY_GITHUB_DSN = os.environ.get("SENTRY_GITHUB_DSN")
-
 
 class GithubSentryError(Exception):
     pass
 
 
-def get(url):
-    headers = {}
-    if GH_TOKEN and url.find("github.com") >= 0:
-        headers["Authorization"] = f"token {GH_TOKEN}"
-    req = requests.get(url, headers=headers)
-    if not req.ok:
-        raise Exception(req.text)
-    return req
+class GithubClient:
+    # This transform GH jobs conclusion keywords to Sentry performance status
+    github_status_trace_status = {"success": "ok", "failure": "internal_error"}
 
+    def __init__(self, dsn=None, github_token=None, dry_run=False) -> None:
+        self.token = github_token
 
-def get_uuid():
-    return uuid.uuid4().hex
+        if not dsn and not dry_run:
```

Comment:
The only reason we don't make `dsn` mandatory is because it allows for easier testing. I should probably changed this in the future.